### ISSUE TITLE
Lifetime issue with Replacer

### ISSUE THREAD

**Issue by ArtemGr** [rust-lang-nursery/regex#242]
Trying to compile the following code

      struct TrReplace;
      impl Replacer for TrReplace {
        fn reg_replace<'a> (&'a mut self, caps: &Captures) -> Cow<'a, str> {
          let text = caps.at (0) .expect ("!caps 0");
          Cow::Borrowed (text)
        }
      }

with `regex 0.1.58` and `rustc 1.9.0-nightly (6e0f2f2f0 2016-03-16)` I get the following error:

```
by_user.rs:1328:27: 1328:29 error: cannot infer an appropriate lifetime for lifetime parameter `'t` due to conflicting requirements [E0495]
by_user.rs:1328           let text = caps.at (0) .expect ("!caps 0");
                                                                   ^~
by_user.rs:1327:9: 1330:10 help: consider using an explicit lifetime parameter as shown: fn reg_replace<'a>(&'a mut self, caps: &Captures<'a>) -> Cow<'a, str>
by_user.rs:1327         fn reg_replace<'a> (&'a mut self, caps: &Captures) -> Cow<'a, str> {
by_user.rs:1328           let text = caps.at (0) .expect ("!caps 0");
by_user.rs:1329           Cow::Borrowed (text)
by_user.rs:1330         }
error: aborting due to previous error
```

I think returning an unmodified string from `reg_replace` is supposed to work, isn't it?

**Comment by BurntSushi** [comment]
I think this is a dupe of #151. See my comment here: https://github.com/rust-lang-nursery/regex/issues/151#issuecomment-172234377 --- Note that this is fixed for the `bytes::Regex` API and will also be fixed when `1.0` is out.

If you're stuck, then I think the best work-around is to probably copy [the `Replacer` trait](https://github.com/rust-lang-nursery/regex/blob/master/src/re_unicode.rs#L717-L767) and [the `replace` methods](https://github.com/rust-lang-nursery/regex/blob/master/src/re_unicode.rs#L561-L608) that use it from the `1.0` branch:

**Comment by ArtemGr** [comment]
Indeed, somehow I missed the #151, thanks.
